cartographer_ros/cartographer_ros/node.cc

```diff
@@ -62,55 +66,50 @@ Node::~Node() {
 
 void Node::Initialize() {
   carto::common::MutexLocker lock(&mutex_);
-  submap_list_publisher_ =
-      node_handle_.advertise<::cartographer_ros_msgs::SubmapList>(
-          kSubmapListTopic, kLatestOnlyPublisherQueueSize);
-  submap_query_server_ = node_handle_.advertiseService(
-      kSubmapQueryServiceName, &Node::HandleSubmapQuery, this);
+  submap_list_publisher_ = node_handle_->create_publisher<::cartographer_ros_msgs::msg::SubmapList>(kSubmapListTopic, rmw_qos_profile_default);
+  submap_query_server_ = node_handle_->create_service<::cartographer_ros_msgs::srv::SubmapQuery>(kSubmapQueryServiceName,
+        [this] (
+                const std::shared_ptr<::cartographer_ros_msgs::srv::SubmapQuery::Request> request,
+                std::shared_ptr<::cartographer_ros_msgs::srv::SubmapQuery::Response> response)
+        {
+          carto::common::MutexLocker lock(&mutex_);
+          map_builder_bridge_.HandleSubmapQuery(request, response);
+        });
 
   if (options_.map_builder_options.use_trajectory_builder_2d()) {
     occupancy_grid_publisher_ =
-        node_handle_.advertise<::nav_msgs::OccupancyGrid>(
-            kOccupancyGridTopic, kLatestOnlyPublisherQueueSize,
-            true /* latched */);
+      node_handle_->create_publisher<::nav_msgs::msg::OccupancyGrid>(kOccupancyGridTopic, rmw_qos_profile_default);
```

Comment:
Yeah, very good point.  I actually haven't gotten these through successfully yet, this may be one reason.  I'll fix this.